### PR TITLE
arch_updates: Add hide_if_zero

### PR DESCRIPTION
**Pull request by lasers**
Make `UPD: 0` go away.

### CODE REVIEW

**Comment by tobes** [comment]
@lasers Thanks for this PR.

The tests are failing because:

* You need to add the new parameter to the docstring
  `hide_if_zero: Do not show if no updates (default False)`

* There is an over indentation error on lines 63 and 64

If you could get the tests to pass that would be great.  If you have problems fixing them just say in this PR

File: py3status/modules/arch_updates.py [hunk 1]
```diff
@@ -11,6 +11,8 @@
     cache_timeout: How often we refresh this module in seconds (default 600)
     format: Display format to use
         (default 'UPD: {pacman}' or 'UPD: {pacman}/{aur}')
+    hide_if_zero: Don't show on bar if False
```
Comment (tobes) [inline py3status/modules/arch_updates.py]: > Don't show on bar if False

Don't show on bar if __True__

File: py3status/modules/README.md [hunk 2]
```diff
@@ -159,6 +159,7 @@ also waiting.
 Configuration parameters:
   - `cache_timeout` How often we refresh this module in seconds *(default 600)*
   - `format` Display format to use
+  - `hide_if_zero` Don't show on bar if True *(default False)*
```
Comment (tobes) [inline py3status/modules/README.md]: These should be in alphabetical order, you've also put it in the middle of the format (before the default).

This file gets auto generated before a release but there is no harm in you adding the updates.

File: py3status/modules/imap.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
     cache_timeout: how often to run this check (default 60)
     criterion: status of emails to check for (default 'UNSEEN')
     format: format to display (default 'Mail: {unseen}')
-    hide_if_zero: don't show on bar if False (default False)
+    hide_if_zero: don't show on bar if True (default False)
```
Comment (tobes) [inline py3status/modules/imap.py]: Good catch :)

Usually it'd be good to do this as a different PR but it's not that important so you might as well leave it here.

File: py3status/modules/README.md (hunk 2 above)
Comment (lasers) [inline py3status/modules/README.md]: Ah rats! I didn't see that. Sorry. Should I fix it?
If not and if that's everything... Well... Thanks for the big help. :-)

**Comment by lasers** [comment]
@tobes Thanks for everything! :-)

**Comment by tobes** [comment]
@lasers cheers for the fix-ups

For what it's worth this can also be achieved by `format = "[\?not_zero UPD: {pacman}]"` but having a user friendly option seems fine to me.  Also the format functionality needs applying to all the modules and documenting, so the typical user will be unaware that this is even possible.

**Comment by ultrabug** [comment]
Thanks a lot for your contribution @lasers ! and @tobes for the review :)

**Comment by lasers** [comment]
@tobes, @ultrabug: It worked well all day with multiple minor Pacman updates. I now notice a minor issue with only AUR updates. When there is an aur_updates, it'll use `__format_pacman_only` instead of `__format_pacman_and_aur`... resulting in `UPD: 0` instead of `UPD: 0/1`. Could somebody take a look at that?

P.S. It might be a nice idea to support `__format_aur_only` too so one could format all three things differently like... `Pacman: 3`, `AUR: 1`, or `Updates: 3/1`.

**Comment by tobes** [comment]
The real issue is #498 the logic is broken
